### PR TITLE
HS-1318 - Expose countAlerts

### PR DESCRIPTION
**Pull request by BenjaminJ**
## Description
- add method to count alerts with possibilyt to filter on severity, expose this in BFF.
- add severity and time filter

## Jira link(s)
- https://issues.opennms.org/browse/HS-1318
- https://issues.opennms.org/browse/HS-1285

## Flagged for review
Branched from HS-1285, not reviewed yet: https://github.com/OpenNMS-Cloud/horizon-stream/pull/901

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts

### CODE REVIEW

**Review by j-white — COMMENTED** [review]
Looks good overall!

File: shared-lib/events/src/main/proto/alerts.proto [hunk 1]
```diff
@@ -134,6 +134,10 @@ message ListAlertsRequest {
   // Optional. A pagination token returned from a previous call to `listAlerts`
   // that indicates where this listing should continue from.
   string next_page_token = 3;
+  string filter = 4;
+  string sortBy = 5;
```
[the remark this answers is not in the document]
Comment (yangli2014) [reply shared-lib/events/src/main/proto/alerts.proto:138]: I prefer the camel case.

File: shared-lib/events/src/main/proto/alerts.proto [hunk 2]
```diff
@@ -134,6 +134,10 @@ message ListAlertsRequest {
   // Optional. A pagination token returned from a previous call to `listAlerts`
   // that indicates where this listing should continue from.
   string next_page_token = 3;
+  string filter = 4;
```
[the remark this answers is not in the document]
Comment (BenjaminJ) [reply shared-lib/events/src/main/proto/alerts.proto]: Should be better now

File: alert/src/main/java/org/opennms/horizon/alertservice/grpc/AlertGrpcService.java [hunk 3]
```diff
@@ -28,44 +28,68 @@
...
+
+        // Get Filters
+        List<Date> timeRange = new ArrayList<>();
+        List<Severity> severities = new ArrayList<>();
+        getFilter(request, timeRange, severities);
+
+        Optional<String> lookupTenantId = tenantLookup.lookupTenantId(Context.current());
+        var alertPage = lookupTenantId
+            .map(tenantId -> alertRepository.findBySeverityInAndLastEventTimeBetweenAndTenantId(severities, timeRange.get(0), timeRange.get(1), pageRequest, tenantId))
+            .orElseThrow();
```
Comment (cgorantla) [inline alert/src/main/java/org/opennms/horizon/alertservice/grpc/AlertGrpcService.java:92]: Do we need to throw DB exceptions outside repository ?  I would opt for catching and returning some error.

**Review by cgorantla — APPROVED** [review]
LGTM!

File: alert/src/main/java/org/opennms/horizon/alertservice/grpc/AlertGrpcService.java (hunk 3 above)
Comment (BenjaminJ) [reply alert/src/main/java/org/opennms/horizon/alertservice/grpc/AlertGrpcService.java:92]: Will be addressed in https://opennms.atlassian.net/browse/HS-1359

**Pull request merged**